packages/@aws-cdk/pipelines/lib/pipeline.ts

```diff
@@ -157,13 +168,17 @@ export class CdkPipeline extends CoreConstruct {
       if (props.crossAccountKeys !== undefined) {
         throw new Error('Cannot set \'crossAccountKeys\' if an existing CodePipeline is given using \'codePipeline\'');
       }
+      if (props.crossAccountKeyRotationEnabled !== undefined) {
+        throw new Error('Cannot set \'crossAccountKeyRotationEnabled\' if an existing CodePipeline is given using \'codePipeline\'');
+      }
```

Comment:
I would also add a check if `crossAccountKeyRotationEnabled` is `true`, but `crossAccountKeys` is `false` - that seems like an error to me.

Reply:
Thought that that check should've added too this morning

Reply:
I see you already added this check to the `Pipeline` construct, which is probably the correct place for it. So, you can remove this entire `if`.